159-part2-iconTags

Bankso opened the pull request:
Adds script that defines a new `iconTag` column. Script takes inputs:
- metadata CSV with relevant annotations, expected to be the QC'd tables derived from `union_qc.py`. Expected usage is to run `union_qc.py`, run `build_tag_lists.py`, upload to designated Synapse folder expected by syncing scripts
- term CSV with columns "term" and "label". Term contains annotations found in the metadata CSV, label contains the categorical terms that will be used for icons on the CCKP. This list is maintained in the data-models repo as modules/shared/mc2_iconTag_map_3-4-25.csv
- annotation types analyzed for tags are defined by attribute_list in main on row 78

I also adjusted the syncing scripts to handle the `iconTags` column, including conversion to string list and inclusion in final database CSVs

Bankso committed:
Create build_tag_lists.py

Script takes inputs:
- metadata CSV with relevant annotations (intended to be used as part of syncing scripts)
- term CSV with columns "term" and "label". Term contains annotations found in the metadata CSV, label contains the categorical terms that will be used for icons on the CCKP. This list is maintained in the data-models repo as modules/shared/mc2_iconTag_map_3-4-25.csv
- annotation types analyzed for tags are defined by attribute_list in main on row 78

Bankso committed:
Add "iconTags" to sync scripts

"iconTags" added for conversion to string list and inclusion in final database CSVs

vpchung commented on portal_tables/build_tag_lists.py:
```diff
@@ -0,0 +1,86 @@
+    syn.login()
+
+    return syn
+
```
Remove if unneeded?

vpchung commented on portal_tables/build_tag_lists.py:
```diff
@@ -0,0 +1,86 @@
+    )
+    parser.add_argument(
+        "-t", help="Path to CSV with annotation-to-tag mappings."
+    )
```
Looks like the script is dependent on these inputs, so I suggest marking these params as required, e.g. `required=True`, or alternatively, give default values when none are given, e.g. `default="syn123"`

Bankso replied:
Yep, they are required, thanks for pointing that out! I think it could make sense to set a `default` for `-t` in the future. I was planning to store the annotation-to-tag mapping in Synapse and pull it via `synapse get` (hence the unused Synapse login function above) but I'm storing in GitHub for now, so I'll just mark them both as required for the time being

vpchung commented on portal_tables/build_tag_lists.py:
```diff
@@ -0,0 +1,86 @@
+            print(f"Accessing annotations for column {column}")
+            annotations = annotations + row[column].split(", ")
+        print(f"Collected annotations: {annotations}")
+        labels = set([str(y) for x, y in term_tuples for n in annotations if n == x])
```
Any chance we can use more descriptive variable names in this comprehension?  I'm not sure what `x` and `y` are, so it's not immediately clear to me what the end result would be.

Bankso committed:
Add requested changes

- removed login function and reference in main
- added required=True for inputs -d and -t
- renamed tuples in list comprehensions for clarity

vpchung reviewed (approved):
LGTM! 🔥
